src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -1381,6 +1390,124 @@ void java_bytecode_parsert::rclass_attribute(classt &parsed_class)
+          // each entry contains a MethodHandle structure
+          // u2 tag
+          // u2 reference kind which must be in the range from 1 to 9
+          // u2 reference index into the constant pool
+          //
+          // reference kinds use the following
+          // 1 to 4 must point to a CONSTANT_Fieldref structure
+          // 5 or 8 must point to a CONSTANT_Methodref structure
+          // 6 or 7 must point to a CONSTANT_Methodref or
+          // CONSTANT_InterfaceMethodref structure, if the class file version
+          // number is 52.0 or above, to a CONSTANT_Methodref only in the case
+          // of less than 52.0
+          // 9 must point to a CONSTANT_InterfaceMethodref structure
+
+          // the index must point to a CONSTANT_String
+          //                           CONSTANT_Class
+          //                           CONSTANT_Integer
+          //                           CONSTANT_Long
+          //                           CONSTANT_Float
+          //                           CONSTANT_Double
+          //                           CONSTANT_MethodHandle
+          //                           CONSTANT_MethodType
+          u2 arg_index1 = read_u2();
+          u2 arg_index2 = read_u2();
+          u2 arg_index3 = read_u2();
+
+          // skip rest
+          for(size_t i = 3; i < num_bootstrap_arguments; i++)
+            read_u2();
+
+          const pool_entryt &arg1 = pool_entry(arg_index1);
+          const pool_entryt &arg2 = pool_entry(arg_index2);
+          const pool_entryt &arg3 = pool_entry(arg_index3);
+
+          if(!(arg1.tag == CONSTANT_MethodType &&
+               arg2.tag == CONSTANT_MethodHandle &&
+               arg3.tag == CONSTANT_MethodType))
+            return;
+
+          lambda_method_handlet real_handle;
+          status() << "INFO: parse lambda handle" << eom;
+          const pool_entryt &lambda_entry = pool_entry(arg_index2);
+          parse_methodhandle(lambda_entry, real_handle);
+          if(
+            real_handle.handle_type !=
+            method_handle_typet::LAMBDA_METHOD_HANDLE)
+          {
+            lambda_method_handlet empty_handle;
+            parsed_class.lambda_method_handle_map[parsed_class.name].push_back(
+              empty_handle);
+            error() << "ERROR: could not parse lambda function method handle"
+                    << eom;
+          }
+          else
+          {
+            real_handle.interface_type = pool_entry(arg1.ref1).s;
+            real_handle.method_type = pool_entry(arg3.ref1).s;
+            parsed_class.lambda_method_handle_map[parsed_class.name].push_back(
+              real_handle);
+            status()
+              << "lambda function reference "
+              << id2string(real_handle.lambda_method_name) << " in class \""
+              << parsed_class.name << "\""
+              << "\n  interface type is "
+              << id2string(real_handle.interface_type = pool_entry(arg1.ref1).s)
```

Comment:
Why are we performing an assignment in the middle of a status output? Wasn't `real_handle.interface_type` assigned to `pool_entry(arg1.ref1).s` back on line 1476.